setup-dev-env.sh

```diff
@@ -3,5 +3,4 @@ set -ex
 
 pip install --upgrade pip
 pip install -r requirements-dev.txt
-pip install -r requirements.txt
-pip install -e .
+pip install -e ".[torch]"
```

Comment:
Since the pytorch module is optional, I'd recommend leaving `pip install -e .` as a default and designing a `setup-dev-env-pytorch.sh` or documenting in [CONTRIBUTING.md#step-by-step](https://github.com/awslabs/aws-data-wrangler/blob/pytorch/CONTRIBUTING.md#step-by-step) how to add it.

Reply:
This is arguable. Should we have everything install for development purpose or selectively install requirements. Since this is only concerns the development process I will resolve and we can discuss and change it later.